state_formatters/dp_formatters.py

```diff
@@ -1223,32 +1376,127 @@ def prompts_goals_collector_formatter(dialog: Dict) -> List[Dict]:
 
 def image_captioning_formatter(dialog: Dict) -> List[Dict]:
     # Used by: image_captioning
-    return [{"image_paths": [dialog["human_utterances"][-1].get("attributes", {}).get("image")]}]
+    return [
+        {
+            "image_paths": [
+                dialog["human_utterances"][-1].get("attributes", {}).get("image")
+            ]
+        }
+    ]
 
 
 def external_integration_skill_formatter(dialog: Dict) -> List[Dict]:
     last_sentences = [dialog["human_utterances"][-1]["text"]]
     dialog_ids = [dialog.get("dialog_id", "unknown")]
     user_ids = [dialog["human_utterances"][-1]["user"]["id"]]
-    return [{"sentences": last_sentences, "dialog_ids": dialog_ids, "user_ids": user_ids}]
+    return [
+        {"sentences": last_sentences, "dialog_ids": dialog_ids, "user_ids": user_ids}
+    ]
 
 
 def robot_formatter(dialog: Dict) -> Dict:
     """This formatter currently provides the JSON as is, without modifying it.
     Either edit it later or choose one of the existing formatters"""
-    detected = get_intents(dialog["human_utterances"][-1], probs=True, which="intent_catcher")
+    detected = get_intents(
+        dialog["human_utterances"][-1], probs=True, which="intent_catcher"
+    )
     return [{"detected": detected}]
 
 
 def dff_command_selector_skill_formatter(dialog: Dict) -> List[Dict]:
-    intents = list(dialog["human_utterances"][-1]["annotations"].get("intent_catcher", {}).keys())
+    intents = list(
+        dialog["human_utterances"][-1]["annotations"].get("intent_catcher", {}).keys()
+    )
     called_intents = {intent: False for intent in intents}
     for utt in dialog["human_utterances"][-5:-1]:
-        called = [intent for intent, value in utt["annotations"].get("intent_catcher", {}).items() if value["detected"]]
+        called = [
+            intent
+            for intent, value in utt["annotations"].get("intent_catcher", {}).items()
+            if value["detected"]
+        ]
         for intent in called:
             called_intents[intent] = True
 
     batches = utils.dff_formatter(dialog, "dff_command_selector_skill")
     batches[-1]["dialog_batch"][-1]["called_intents"] = called_intents
     batches[-1]["dialog_batch"][-1]["dialog_id"] = dialog.get("dialog_id", "unknown")
     return batches
+
+
+def user_emotion_bot_mood_formatter(dialog: Dict) -> List[Dict]:
+    # comment out the next two lines to save dialog data to see its contents
+    # with open('test_formatters.json', 'w', encoding='utf-8') as f:
+    #     json.dump(dialog, f, ensure_ascii=False, indent=4)
+    sentences = [dialog["human_utterances"][-1]["text"]]
+
+    user_emotions = (
+        dialog["human_utterances"][-1]["annotations"]
+        .get("combined_classification", {})
+        .get("emotion_classification", {})
```

Comment:
why do you do this pre-processing in formatter?
I suggest to move it inside your annotator. And use common.utils.get_emotion, common.utils.get_sentiment functions to have a backward compatibility.

Reply:
moved preprocessing of sentiment and emotion to annotator, used functions from common.utils